langlib/lang.error/src/main/java/org/ballerinalang/langlib/error/StackTrace.java

```diff
@@ -77,15 +81,38 @@ private static BArray getCallStackArray(StackTraceElement[] stackTrace) {
     }
 
     static BMap<BString, Object> getStackFrame(StackTraceElement stackTraceElement) {
+        String methodName = stackTraceElement.getMethodName();
+        String moduleName = IdentifierUtils.decodeIdentifier(stackTraceElement.getClassName());
+        String fileName = stackTraceElement.getFileName();
+        int lineNumber = stackTraceElement.getLineNumber();
+
         Object[] values = new Object[4];
-        values[0] = stackTraceElement.getMethodName();
-        values[1] = stackTraceElement.getClassName();
-        values[2] = stackTraceElement.getFileName();
-        values[3] = stackTraceElement.getLineNumber();
+        values[0] = methodName;
+        values[2] = fileName;
+        values[3] = lineNumber;
+        if (!isSingleBalFile(moduleName, fileName)) {
```

Comment:
module name and file name can be replaced twice during isSingleBalFile and getModuleName methods.

`
      String moduleName = IdentifierUtils.decodeIdentifier(stackTraceElement.getClassName())
                .replace(FILE_NAME_PERIOD_SEPARATOR, DOT);
        String fileName = stackTraceElement.getFileName().replace(BLANG_SRC_FILE_SUFFIX, "");
        if (!moduleName.equals(fileName)) {
            int index = moduleName.lastIndexOf("." + fileName);
            if (index != -1) {
                values[1] = moduleName.substring(0, index);
            } else {
                values[1] = getModuleName(moduleName, fileName);
            }
        }

`

Reply:
I also had the same concern, IMO, since the `isSingleBalFile` method returns only a boolean value the changes on the `moduleName` or `fileName` will not be reflected inside the `getStackFrame`. But rewriting as per your suggestion will be a better way than previous. I'll address it with an upcoming commit.